source/fuzz/fact_manager/data_synonym_and_id_equation_facts.cpp

```diff
@@ -53,11 +53,15 @@ bool DataSynonymAndIdEquationFacts::OperationEquals::operator()(
 
 void DataSynonymAndIdEquationFacts::AddFact(
     const protobufs::FactDataSynonym& fact,
+    const DeadBlockFacts& dead_block_facts,
     const IrrelevantValueFacts& irrelevant_value_facts,
```

Comment:
This looks a bit convoluted to me: we are not using `dead_block_facts` anywhere in the method and only passing it down into `irrelevant_value_facts.IdIsIrrelevant` call. Maybe it's better to refactor all submanagers into class fields instead? @afd, @stefanomil, what do you think?

Reply:
Summary of the discussion: we will keep it as it is (at least for now).